pkg/apis/flyteworkflow/v1alpha1/workflow.go

```diff
@@ -222,6 +226,10 @@ func (in *WorkflowSpec) FromNode(name NodeID) ([]NodeID, error) {
 	return downstreamNodes, nil
 }
 
+func (in *WorkflowSpec) GetOnFailurePolicy() core.WorkflowMetadata_OnFailurePolicy {
```

Comment:
we need to use jsonpb to deserialize.

Reply:
Do we need jsonpb? this should work, right?